File: cs/number3.cs
using System;

public class CountSeriesFuncNoComments
{
    public static int CntNonDecSeries()
    {
        int len_series = 0;
        int cnt = 0;

        int len = int.Parse(Console.ReadLine());

        if (len > 0)
        {
            int num1 = int.Parse(Console.ReadLine());
            int num2 = 0;

            for (int i = 1; i < len; i++)
            {
                num2 = int.Parse(Console.ReadLine());
                if (num2 >= num1)
                {
                    len_series++;
                }
                else if (len_series != 0)
                {
                    cnt++;
                    len_series = 0;
                }
                num1 = num2;
            }

            if (len_series != 0)
            {
                cnt++;
            }
        }
        return cnt;
    }

    public static void Main(string[] args)
    {
        int result = CntNonDecSeries();
        Console.WriteLine($"Количество неубывающих серий: {result}");
    }
}

File: rs/number3.rs
use std::io;

fn cnt_non_dec_series() -> i32 {
    let mut len_series = 0;
    let mut cnt = 0;

    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let len: usize = input.trim().parse().unwrap();

    let mut num1 = {
        let mut input = String::new();
        io::stdin().read_line(&mut input).unwrap();
        input.trim().parse::<i32>().unwrap()
    };

    let mut i = 1;
    while i < len {
        let mut input = String::new();
        io::stdin().read_line(&mut input).unwrap();
        let num2 = input.trim().parse::<i32>().unwrap();

        if num2 >= num1 {
            len_series += 1;
            num1 = num2;
        } else if len_series != 0 {
            cnt += 1;
            len_series = 0;
        }
        i += 1;
    }

    if len_series != 0 {
        cnt += 1;
    }

    cnt
}

fn main() {
    println!("Введите количество чисел и саму последовательность:");
    println!("Количество неубывающих серий: {}", cnt_non_dec_series());
}